101-vm-ws2016-visualstudio2019/azuredeploy.json

```diff
@@ -0,0 +1,201 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "vmName": {
+            "type": "string",
+            "defaultValue": "WinSvVisualVM",
+            "metadata": {
+                "description": "The name of you Virtual Machine."
+            }
+        },
+        "VmSize": {
+            "type": "string",
+            "allowedValues": [
+                "Standard_DS1_v2",
+                "Standard_DS2_v2",
+                "Standard_D1_v2",
+                "Standard_D2_v2"
+        ],
+            "metadata": {
+                "description": "The size of the VM"
+            },
+        "defaultValue": "Standard_D2_v2"
+        },
+            "adminUsername": {
+            "type": "string",
+            "metadata": {
+                "description": "Username for the Virtual Machine."
+            }
+        },
+        "adminPassword": {
+            "type": "secureString",
+            "metadata": {
+                "description": "Password for the Virtual Machine. The password must be at least 12 characters long and have lower case, upper characters, digit and a special character (Regex match)"
+            }
+        },
+        "location": {
+        "type": "string",
+        "defaultValue": "[resourceGroup().location]",
+        "metadata": {
+                "description": "Location for all resources."
+            }
+        }
+    },
+    "variables": {
+        "location": "[parameters('location')]",
+        "virtualNetworkName": "[concat('MyVNet-',  parameters('vmName'))]",
+        "publicIpAddressName": "[concat('myPublicIP-',  parameters('vmName'))]",
+        "subnetName": "[concat('mySubnet-',  parameters('vmName'))]",
+        "networkInterfaceName": "[concat('myNetInt-',  parameters('vmName'))]",
+        "networkSecurityGroupName": "[concat('mySecGroup-',  parameters('vmName'))]",       
+        "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]",
+        "osDiskType": "StandardSSD_LRS"
+         
+    },
+    "resources": [
+        {
+            "name": "[variables('networkInterfaceName')]",
+            "type": "Microsoft.Network/networkInterfaces",
+            "apiVersion": "2018-11-01",
+            "location": "[variables('location')]",
+            "dependsOn": [
```

Comment:
remove concat and use resourceId throughout

Reply:
It makes more sense. 
It's done, thanks!